keyboard.html

```diff
@@ -0,0 +1,97 @@
+<!DOCTYPE html>
+<html>
+<head>
+  <title>||Working with elements||</title>
+  <style>
+    #keyboard {
+      display: flex;
+    }
+    .white-key {
+      height: 100px;
+      width: 28px;
+      background-color: white;
+      outline-style: solid;
+      outline-width: thin
+    }
+
+    .black-key {
+      height: 60px;
+      background-color: black;
+      position: absolute;
+      outline-style: solid;
+      outline-width: thin
+    }
+  </style>
+</head>
+<body>
+  <div id="description"></div>
+  <div id="keyboard"></div>
+  <script> 
+    const keyboard = document.getElementById('keyboard')
+
+    document.body.onload = addKeys(14);
+    window.addEventListener('resize', resizeAllBlackKeys)
+    const allBlackKeysArray = Array.from(document.getElementsByClassName('black-key'))
+    
+    function addKeys(numKeys) {
+      const whiteKeysPerOctave = 6
+      let i = 0;
+      let whiteKeyCount = 0;
+      while(i < numKeys) {
+        // create a new rectangle div element
+        const newDiv = document.createElement("div");
+        newDiv.setAttribute('id', `white-key${i}`)
+        newDiv.classList.add('white-key')
+    
+        // add the newly created element and its content into the DOM
+        keyboard.appendChild(newDiv);
+
+        var coords = getPos(document.querySelector(`#white-key${i}`));
+
+        if(whiteKeyCount < whiteKeysPerOctave) {
+          if(blackKeyCheck(whiteKeyCount)) {
+            addBlackKey();
+          }
+          whiteKeyCount++
+        }
+        else {
+          whiteKeyCount = 0 
+        }
+          
+        i++
+      }
+
+      const totalKeyCount = keyboard.childElementCount;
+      document.getElementById("description").innerHTML = `Below should be a keyboard with ${totalKeyCount} keys.`;
+    }
+
+    function addBlackKey() {
+      const newDiv = document.createElement("div");
+      newDiv.classList.add('black-key')
+      keyboard.appendChild(newDiv);
+      sizeBlackKey(newDiv)
+    }
+
+    function getPos(el) {
+      var rect=el.getBoundingClientRect();
+      return rect
+    }
+
+    function blackKeyCheck(whiteKeyCount) {
+      const whiteKeysNeedingBlackKeys = [0,1,3,4,5]
+      return whiteKeysNeedingBlackKeys.includes(whiteKeyCount)
+    }
+
+    function resizeAllBlackKeys() {
+      allBlackKeysArray.forEach(sizeBlackKey)
+    }
+
+    function sizeBlackKey(blackKey) {
+      const whiteKeySibling = blackKey.previousElementSibling
+      const whiteKeySiblingCoords = getPos(whiteKeySibling)
```

Comment:
for performance you could calculate this once and then pass it in, since all white keys should be the same size